include/gsl/gsl_assert

```diff
@@ -68,17 +68,32 @@
 
 namespace gsl
 {
+
 struct fail_fast : public std::runtime_error
 {
     explicit fail_fast(char const* const message) : std::runtime_error(message) {}
 };
-}
+
+template <typename Exception>
+void throw_exception(Exception&& exception);
```

Comment:
Doesn't this also need an implementation that does std::terminate() in the case that some compiler flag that specifies no-exceptions mode is on?